cirq-core/cirq/protocols/decompose_protocol_test.py

```diff
@@ -398,3 +398,44 @@ def test_decompose_recursive_dfs(with_context: bool):
             if with_context
             else mock_qm.method_calls == expected_calls
         )
+
+
+class G1(cirq.Gate):
+    def _num_qubits_(self) -> int:
+        return 1
+
+    def _decompose_with_context_(self, qubits, context):
+        yield cirq.CNOT(qubits[0], context.qubit_manager.qalloc(1)[0])
+
+
+class G2(cirq.Gate):
+    def _num_qubits_(self) -> int:
+        return 1
+
+    def _decompose_with_context_(self, qubits, context):
+        yield G1()(*context.qubit_manager.qalloc(1))
+
+
+def test_successive_decompose_once_succeed():
+    from cirq.protocols.decompose_protocol import _reset_context_counter
```

Comment:
nit - maybe we can use mocking for `_CONTEXT_COUNTER` and remove the `_reset_context_counter()` as it is only called from the test.

```suggestion
@mock.patch('cirq.protocols.decompose_protocol._CONTEXT_COUNTER', itertools.count())
def test_successive_decompose_once_succeed():
```

Reply:
done